sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -0,0 +1,122 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Sampler;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Builder;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanData;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** {@link SpanBuilderSdk} is SDK implementation of {@link Span.Builder}. */
+class SpanBuilderSdk implements Span.Builder {
+
+  private SpanContext parent;
+  private Kind spanKind;
+  private List<Link> links;
+  private boolean recordEvents;
+  private Sampler sampler;
+  private final String spanName;
+  private ParentType parentType = ParentType.CURRENT_SPAN;
+
+  SpanBuilderSdk(String spanName) {
+    this.spanName = spanName;
+  }
+
+  @Override
+  public Builder setParent(Span parent) {
+    Utils.checkNotNull(parent, "parent");
+    this.setParent(parent.getContext());
+    return this;
+  }
+
+  @Override
+  public Builder setParent(SpanContext remoteParent) {
+    this.parent = Utils.checkNotNull(remoteParent, "remoteParent");
+    this.parentType = ParentType.PARENT;
+    return this;
+  }
+
+  @Override
+  public Builder setNoParent() {
+    this.parentType = ParentType.NO_PARENT;
+    this.parent = null;
+    return this;
+  }
+
+  @Override
+  public Builder setSpanKind(Kind spanKind) {
+    this.spanKind = Utils.checkNotNull(spanKind, "spanKind");
+    return this;
+  }
+
+  @Override
+  public Builder setSampler(Sampler sampler) {
+    this.sampler = Utils.checkNotNull(sampler, "sampler");
+    return this;
+  }
+
+  @Override
+  public Builder addLink(SpanContext spanContext) {
+    Utils.checkNotNull(spanContext, "spanContext");
+    addLink(SpanData.Link.create(spanContext));
+    return this;
+  }
+
+  @Override
+  public Builder addLink(SpanContext spanContext, Map<String, AttributeValue> attributes) {
+    Utils.checkNotNull(spanContext, "spanContext");
+    Utils.checkNotNull(attributes, "attributes");
+    addLink(SpanData.Link.create(spanContext, attributes));
+    return this;
+  }
+
+  @Override
+  public Builder addLink(Link link) {
+    Utils.checkNotNull(link, "link");
+    if (links == null) {
+      links = new ArrayList<>();
+    }
+    links.add(link);
+    return this;
+  }
+
+  @Override
+  public Builder setRecordEvents(boolean recordEvents) {
+    this.recordEvents = recordEvents;
+    return this;
+  }
+
+  @Override
+  public SpanSdk startSpan() {
+    // TODO get parent span from the context if noParent=false and parent/remoteParents are null
+    return null;
+  }
+
+  private enum ParentType {
+    CURRENT_SPAN,
+    PARENT,
+    NO_PARENT,
```

Comment:
I'd find `DEFAULT_PARENT`, `EXPLICIT_PARENT` and `NO_PARENT` more descriptive but I'm fine with the current naming as well.

Reply:
+1

Reply:
+1 on `EXPLICIT_PARENT`. I will keep the `CURRENT_SPAN` as it better matches the used terminology in the project.